domain/secretbackend/state/state.go

```diff
@@ -61,13 +72,10 @@ WHERE  uuid = $M.uuid`, sqlair.M{}, ModelSecretBackend{})
 		return secretbackend.ModelSecretBackend{}, errors.Trace(err)
 	}
 	var m ModelSecretBackend
-	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		err := tx.Query(ctx, stmt, sqlair.M{"uuid": uuid}).Get(&m)
-		if errors.Is(err, sql.ErrNoRows) {
-			return fmt.Errorf("%w: %q", modelerrors.NotFound, uuid)
-		}
-		return errors.Trace(err)
-	})
+	err = tx.Query(ctx, stmt, sqlair.M{"uuid": uuid}).Get(&m)
+	if errors.Is(err, sql.ErrNoRows) {
+		return secretbackend.ModelSecretBackend{}, fmt.Errorf("%w: %q", modelerrors.NotFound, uuid)
```

Comment:
This error can be written a little better to give more context. `fmt.Errorf("cannot get secret backend for model %q: %w", uuid, model errors.NotFound)`